packages/organization-viewer-cli/src/index.ts

```diff
@@ -1,65 +1,114 @@
+
 // import data from './org-data.json'
 import {
   aragonConnect,
   Permission,
-  App
+  App,
+  Organization
 } from 'plumbery-core'
-import Connection from 'plumbery-core/dist/Connection'
+import {
+  Voting,
+  Vote,
+  Cast,
+  GraphQLWrapper
+} from 'plumbery-connector-thegraph'
+import gql from 'graphql-tag'
 
 const ORG_ADDRESS = '0x00e45b9918297037fe6585c2a1e53e8801f562f4'
 
-let app
-
 async function main() {
-  const connection = initConnection()
+  const org = await initAndGetOrg()
+
+  await inspectOrg(org)
+
+  await inspectVotingHighLevel(org)
+  await inspectVotingLowLevel(org)
+}
+
+async function initAndGetOrg(): Promise<Organization> {
+  const connection = aragonConnect({
+    connector: [
+      'thegraph',
+      { daoSubgraphUrl: 'https://api.thegraph.com/subgraphs/name/0xgabi/dao-subgraph-rinkeby' }
+    ],
+    signer: {},
+  })
+
+  console.log('\nOrganization:')
   const org = connection.organization(ORG_ADDRESS)
+  console.log(org.toString())
+
+  return org
+}
 
+async function inspectOrg(org: Organization): Promise<void> {
   console.log('\nPermissions:')
   const permissions = await org.permissions()
   permissions.map((permission: Permission) => console.log(permission.toString()))
 
+  console.log('\nA role from a permission:')
+  const role = await permissions[4].getRole()
+  console.log(role.toString())
+
   console.log('\nApps:')
   const apps = await org.apps()
   apps.map((app: App) => console.log(app.toString()))
 
   console.log('\nA voting app:')
-  app = apps.find((app: App) => app.name == 'voting')!
-  console.log(app.toString())
+  const votingApp = apps.find((app: App) => app.name == 'voting')!
+  console.log(votingApp.toString())
 
   console.log('\nA repo from an app:')
-  const repo = await app.repo()
+  const repo = await apps[2].repo()
   console.log(repo.toString())
 
   console.log('\nAn app by address:')
-  app = await org.app(apps[1].address)
-  console.log(app.toString())
+  const appByAddress = await org.app(apps[1].address)
+  console.log(appByAddress.toString())
 
   console.log('\nAn app from a permission:')
-  app = await permissions[1].getApp()
-  if (app) { console.log(app.toString()) }
+  const appFromPermission = await permissions[1].getApp()
+  if (appFromPermission) { console.log(appFromPermission.toString()) }
+}
 
-  console.log('\nA role from a permission:')
-  const role = await permissions[1].getRole()
-  console.log(role.toString())
+async function inspectVotingHighLevel(org: Organization): Promise<void> {
+  const apps = await org.apps()
+  const app = apps.find((app: App) => app.name == 'voting')!
+
+  console.log('\nVoting:')
+  const voting = new Voting(app, 'https://api.thegraph.com/subgraphs/name/ajsantander/voting-subgraph')
```

Comment:
I think we would want an app subgraph encompassing all known apps of one type (or more).

On instantiating app wrappers, perhaps being verbose and forcing users to specify a config here would be good? Otherwise it might be confusing to have some apps available by default but not others (and not really knowing which are the defaults without checking documentation).

Reply:
The "voting" object you refer to above is of type App. Even if we had a generic ".state" property, it wouldn't know anything about ".votes()". What you're asking for is a mixture of experiments 1 & 2 for app wrappers and precisely what this experiment 3 is not trying to do.

Regarding one subgraph for all voting apps, or one subgraph for all "official" apps, I think this deserves a more extensive discussion, because it has its cons.